### PR TITLE
TT-1 subtask TT-24 adding code for message model, schema, interfaces,…

### PR DESCRIPTION
**Pull request by galen94**
… dao

### CODE REVIEW

File: models/messages/Message.ts [hunk 1]
```diff
@@ -12,9 +12,9 @@ import mongoose from "mongoose";
  * @property {Date} sentOn date on which the message was sent
  */
 export default interface Message {
-    _id?: mongoose.Schema.Types.ObjectId,
```
Comment (sghosalkar) [inline models/messages/Message.ts]: We will need this _id for deleting and updating messages

Comment (sghosalkar) [inline models/messages/Message.ts]: We will need this _id for deleting and updating messages

File: daos/message-dao.ts [hunk 2]
```diff
@@ -24,65 +25,119 @@ export default class MessageDao implements MessageDaoI {
...
     }
-
     private constructor() {}
 
     /**
-     * Uses MessageModel to retrieve all message documents received by the user
+     * Uses MessageModel to retrieve all message documents
+     * @returns Promise To be notified when the messages are retrieved from database
+     */
+    findAllMessages = async (): Promise<Message[]> =>
```
Comment (sghosalkar) [inline daos/message-dao.ts:34]: We might never use this

Comment (sghosalkar) [inline daos/message-dao.ts:34]: We might never use this

File: mongoose/messages/MessageSchema.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @file Implements mongoose schema
```
Comment (sghosalkar) [inline mongoose/messages/MessageSchema.ts]: Please rename file as "message-schema.ts" or edit the existing file

Comment (sghosalkar) [inline mongoose/messages/MessageSchema.ts]: Please rename file as "message-schema.ts" or edit the existing file

File: mongoose/messages/MessageModel.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import mongoose from "mongoose";
```
Comment (sghosalkar) [inline mongoose/messages/MessageModel.ts]: Please rename file as "message-model.ts" or edit the existing file

Comment (sghosalkar) [inline mongoose/messages/MessageModel.ts]: Please rename file as "message-model.ts" or edit the existing file

File: mongoose/messages/MessageSchema.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ * for the messages collection
+ */
+
+import mongoose, {Schema} from "mongoose";
+import Message from "../../models/messages/Message";
+
+const MessageSchema = new mongoose.Schema<Message>({
+    message: {type: String, required: true},
+    sender: {type: Schema.Types.ObjectId, ref: "UserModel"},
+    recipient: {type: Schema.Types.ObjectId, ref: "UserModel"},
```
Comment (sghosalkar) [inline mongoose/messages/MessageSchema.ts]: Please add "required: true" validation

Comment (sghosalkar) [inline mongoose/messages/MessageSchema.ts]: Please add "required: true" validation for both sender and receiver

File: models/messages/AttachmentType.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+enum AttachmentType {
```
Comment (sghosalkar) [inline models/messages/AttachmentType.ts]: Please use kebab case for naming files

Comment (sghosalkar) [inline models/messages/AttachmentType.ts]: Please use kebab case for naming files

File: daos/message-dao.ts [hunk 7]
```diff
@@ -24,65 +25,119 @@ export default class MessageDao implements MessageDaoI {
...
-     * @param {string} uid User's primary key
+     * @param {string} uidSender sender's primary key
+     * @param {string} uidRecipient recipient's primary key
      * @param {Message} message Instance to be inserted into the database
      * @returns Promise To be notified when message is inserted into the database
      */
-    sendMessage = async (uid: string, message: Message): Promise<any> => {
-        return MessageModel.create({...message, from: uid});
-    }
+    userSendsMessage = async (uidSender: string, uidRecipient: string, message: Message): Promise<Message> =>
```
Comment (sghosalkar) [inline daos/message-dao.ts]: We can set the values of sender and receiver in Message object

Comment (sghosalkar) [inline daos/message-dao.ts]: We can set the values of sender and receiver in Message object

Comment (galen94) [reply daos/message-dao.ts]: Ok, but then there is no reason to even pass in the uid, so can we just have message as the only parameter?

**Comment by sghosalkar** [comment]
Looks good!

**Review by sghosalkar — APPROVED** [review]

**Review by sanatdeshpande1 — APPROVED** [review]
Looks good!

**Pull request merged**